atc/exec/task_step.go

```diff
@@ -190,16 +195,57 @@ func (step *TaskStep) Run(ctx context.Context, state RunState) error {
 
 	owner := db.NewBuildStepContainerOwner(step.metadata.BuildID, step.planID, step.metadata.TeamID)
 
-	chosenWorker, err := step.workerPool.FindOrChooseWorkerForContainer(
-		ctx,
-		logger,
-		owner,
-		containerSpec,
-		workerSpec,
-		step.strategy,
-	)
-	if err != nil {
-		return err
+	var chosenWorker worker.Worker
+	var activeTasksLock lock.Lock
+
+	for {
+		if step.strategy.ModifyActiveTasks() {
```

Comment:
Something minor, Can we go with with `strategy.ModifiesActiveTasks()`?

Reply:
:+1: